### PR TITLE
RE2022-240: Merge column specs for collection data products

### PR DESCRIPTION
**Pull request by MrCreosote**

### CODE REVIEW

File: src/common/collection_column_specs/load_specs.py [hunk 1]
```diff
@@ -10,13 +10,34 @@
 """
 import yaml
 from pathlib import Path
-from src.common.product_models.columnar_attribs_common_models import ColumnarAttributesSpec
-
+from src.common.product_models.columnar_attribs_common_models import (
+    ColumnarAttributesSpec,
```
Comment (Xiangs18) [inline src/common/collection_column_specs/load_specs.py:14]: Should be alphabetical order

File: src/common/collection_column_specs/load_specs.py [hunk 2]
```diff
@@ -10,13 +10,34 @@
...
+    AttributesColumnSpec,
```
Comment (Xiangs18) [inline src/common/collection_column_specs/load_specs.py:15]: Should be alphabetical order

Comment (MrCreosote) [reply src/common/collection_column_specs/load_specs.py]: fixed

File: src/common/collection_column_specs/load_specs.py [hunk 3]
```diff
@@ -26,8 +47,29 @@ def load_spec(data_product: str, collection: str = None) -> ColumnarAttributesSp
...
+    columns = []
+    for k in keys:
+        first = None
+        for coll, spec in coll2spec.items():
+            if k in spec:
+                if not first:
+                    first = (coll, spec[k])
+                else:
+                    if spec[k] != first[1]:
+                        colls = sorted([coll, first[0]])
```
Comment (Xiangs18) [inline src/common/collection_column_specs/load_specs.py:71]: Not quite sure why you are sorting here. Shouldn't raise error directly?

Comment (MrCreosote) [reply src/common/collection_column_specs/load_specs.py:71]: If I don't sort, then which coll name goes where in the error message is indeterminate, and the test passing is therefore indeterminate. The lack of a sort is why the test were passing locally but not in GHA

**Review by Xiangs18 — APPROVED** [review]
LGTM!

**Pull request merged**